fx-src/FirefoxStorage.js

```diff
@@ -197,6 +202,39 @@ export default class FirefoxAdapter extends BaseAdapter {
     });
   }
 
+  loadDump(records) {
+    let connection = this._connection;
+    let collection_name = this.collection;
+    return Task.spawn(function* () {
+      yield connection.executeTransaction(function* doImport() {
+        for (let record of records) {
+          const params = {
+            collection_name: collection_name,
+            record_id: record.id,
+            record: JSON.stringify(record)
+          };
+          yield connection.execute(statements.importData, params);
+        }
+        const lastModified = Math.max(...records.map(record => record.last_modified));
+        const params = {
+          collection_name: collection_name
+        };
+        const previousLastModified = yield connection.execute(
+          statements.getLastModified, params).then(result => {
+            return result ? result[0].getResultByName('last_modified') : -1;
+          });
+        if (lastModified > previousLastModified) {
+          const params = {
+            collection_name: collection_name,
+            last_modified: lastModified
+          };
+          yield connection.execute(statements.saveLastModified, params);
```

Comment:
> make sure my ``last_modified`` time matches the data I'm storing

That's not exactly what we are doing there.
We are keeping the ``collection_timestamp`` from the last sync except if the records we are importing are more recent that the last sync.